Wikipedia/Code/NotificationsCenterCellViewModel+SheetActionExtensions.swift

```diff
@@ -0,0 +1,445 @@
+
+import Foundation
+
+extension NotificationsCenterCellViewModel {
+    
+    enum SheetAction {
+        case markAsReadOrUnread(SheetActionData)
+        case custom(SheetActionData)
+        case notificationSubscriptionSettings(SheetActionData)
+    }
+    
+    struct SheetActionData {
+        let text: String
+        let url: URL?
+    }
+    
+    func sheetActions(for configuration: Configuration) -> [SheetAction] {
+        
+        var sheetActions: [SheetAction] = []
+        let markAsReadText = WMFLocalizedString("notifications-center-mark-as-read", value: "Mark as Read", comment: "Button text in Notifications Center to mark a notification as read.")
+        let markAsUnreadText = WMFLocalizedString("notifications-center-mark-as-unread", value: "Mark as Unread", comment: "Button text in Notifications Center to mark a notification as unread.")
+        let markAsReadOrUnreadText = isRead ? markAsUnreadText : markAsReadText
+        let markAsReadOrUnreadActionData = SheetActionData(text: markAsReadOrUnreadText, url: nil)
+        sheetActions.append(.markAsReadOrUnread(markAsReadOrUnreadActionData))
+        
+        switch notification.type {
+        case .userTalkPageMessage:
+            sheetActions.append(contentsOf: userTalkPageActions(for: configuration))
+        case .mentionInTalkPage,
+             .editReverted:
+            sheetActions.append(contentsOf: mentionInTalkAndEditRevertedPageActions(for: configuration))
+        case .mentionInEditSummary: //done
```

Comment:
Minor: leftover `//done` comment, unless I'm conceptually missing something about its purpose.

Reply:
@staykids Good catch, those were testing comments to keep track of notification types I was testing. This is removed.